### PR TITLE
WIP: Initial windows test app

### PR DESCRIPTION
**Pull request by AnastasiaOrishchenko**
- Add app skeleton

- TODO: make sure app launches with ARM build config

### CODE REVIEW

File: windows/ReactNativeTestApp/App.cpp [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+/// <summary>
+/// Invoked when the application is launched normally by the end user.  Other entry points
+/// will be used such as when the application is launched to open a specific file.
+/// </summary>
+/// <param name="e">Details about the launch request and process.</param>
+void App::OnLaunched(LaunchActivatedEventArgs const& e)
+{
+    Frame rootFrame{ nullptr };
+    auto content = Window::Current().Content();
+    if (content)
```
Comment (tido64) [inline windows/ReactNativeTestApp/App.cpp]: We should be explicit with our comparisons:
```suggestion
    if (content != nullptr)
```

File: windows/ReactNativeTestApp/App.cpp [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+}
+
+/// <summary>
+/// Invoked when the application is launched normally by the end user.  Other entry points
+/// will be used such as when the application is launched to open a specific file.
+/// </summary>
+/// <param name="e">Details about the launch request and process.</param>
+void App::OnLaunched(LaunchActivatedEventArgs const& e)
+{
+    Frame rootFrame{ nullptr };
```
Comment (tido64) [inline windows/ReactNativeTestApp/App.cpp]: Code style nit:
```suggestion
    Frame rootFrame = nullptr;
```

File: windows/ReactNativeTestApp/App.cpp [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+        rootFrame.NavigationFailed({ this, &App::OnNavigationFailed });
+
+        if (e.PreviousExecutionState() == ApplicationExecutionState::Terminated)
+        {
+            // Restore the saved session state only when appropriate, scheduling the
+            // final launch steps after the restore is complete
+        }
+
+        if (e.PrelaunchActivated() == false)
```
Comment (tido64) [inline windows/ReactNativeTestApp/App.cpp]: Don't compare against booleans.
```suggestion
        if (!e.PrelaunchActivated())
```

File: windows/ReactNativeTestApp/App.cpp [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+            if (rootFrame.Content() == nullptr)
+            {
+                // When the navigation stack isn't restored navigate to the first page,
+                // configuring the new page by passing required information as a navigation
+                // parameter
+                rootFrame.Navigate(xaml_typename<ReactNativeTestApp::MainPage>(), box_value(e.Arguments()));
+            }
+            // Ensure the current window is active
+            Window::Current().Activate();
+        }
```
Comment (tido64) [inline windows/ReactNativeTestApp/App.cpp]: Is this a copy of the block above? Should we extract it to a function?

File: .gitignore [hunk 5]
```diff
@@ -14,3 +14,4 @@ package-lock.json
 report.[0-9]*.[0-9]*.[0-9]*.[0-9]*.json
 xcuserdata/
 yarn-error.log*
+.vs/
```
Comment (tido64) [inline .gitignore:17]: Do we still need this line if it's already specified in `windows/.gitignore`?

**Review by tido64 — COMMENTED** [review]
In general, this looks fine to me. Shouldn't this PR have gone to upstream though?

File: .gitignore (hunk 5 above)
Comment (AnastasiaOrishchenko) [reply .gitignore:17]: Seems like it's needed. Will delete from windows/.gitignore and keen only here.

File: windows/ReactNativeTestApp/App.cpp (hunk 3 above)
Comment (AnastasiaOrishchenko) [reply windows/ReactNativeTestApp/App.cpp]: I should have checked generated code more thoroughly.

File: windows/ReactNativeTestApp/App.cpp (hunk 4 above)
Comment (AnastasiaOrishchenko) [reply windows/ReactNativeTestApp/App.cpp]: Not exactly the same, but can extract part of it.

**Pull request closed without merging**